api/controllers/board.js

```diff
@@ -81,20 +81,45 @@ async function getPublicBoards(req, res) {
 
 async function deleteBoard(req, res) {
   const id = req.swagger.params.id.value;
-  Board.findByIdAndRemove(id, function (err, boards) {
+  const board = Board.findOne({ _id: id }, function (err, board) {
+    if (err) {
+      return res.status(500).json({
+        message: 'Error getting board. ',
+        error: err.message
+      });
+    }
+    if (!board) {
+      return res.status(404).json({
+        message: 'Board does not exist. Board Id: ' + id
+      });
+    }
+    return board;
+  });
+  //  Validate id
+  if (!ObjectId.isValid(id)) {
+    return res.status(404).json({
+      message: 'Invalid ID for a Board. Board Id: ' + id
+    });
+  }
+  if (!req.user.isAdmin && req.user !== board.author) {

```

Comment:
Personally I would return a 404 rather than a 403 even though this is related to authorization. One advantage to returning a 404 is that it doesn't tell the caller that the board actually exists, which seems safer considering other routes might still be vulnerable.